### DISCUSSION TITLE
Cross compile of the pqxx in c++ for arm7

### DISCUSSION THREAD

**Discussion by ghost** [jtv/libpqxx#864] [General]
I want to use the library libpqxx in my plc next project, and i used the toolchain of the plc next and after i followed the instructions how to build my library:

$env:ARP_TOOLCHAIN_ROOT = "C:/SDKs/AXCF2152/2024.0"

cd C:\Libraries\libpqxx\build_arm cmake -DCMAKE_TOOLCHAIN_FILE=C:/SDKs/AXCF2152/2024.0/toolchain.cmake -DCMAKE_INSTALL_PREFIX=C:/Libraries/libpqxx_arm -DPostgreSQL_INCLUDE_DIR="C:/Program Files/PostgreSQL/16/include" -DPostgreSQL_LIBRARY="C:/Program Files/PostgreSQL/16/lib/libpq.lib" ..

cmake --build . --config Release

After the last command i got this at the end:

libpqxx-7.9.a(connection.obj) : error LNK2019: unresolved external symbol __imp_ioctlsocket referenced in function "pub lic: void __cdecl pqxx::connection::set_blocking(bool)& " (?set_blocking@connection@pqxx@@QEGAAX_N@Z) [C:\Libraries\lib pqxx\build_arm\test\runner.vcxproj] libpqxx-7.9.a(connection.obj) : error LNK2019: unresolved external symbol __imp_WSAGetLastError referenced in function "public: void __cdecl pqxx::connection::set_blocking(bool)& " (?set_blocking@connection@pqxx@@QEGAAX_N@Z) [C:\Libraries \libpqxx\build_arm\test\runner.vcxproj] libpqxx-7.9.a(wait.obj) : error LNK2001: unresolved external symbol __imp_WSAGetLastError [C:\Libraries\libpqxx\build_a rm\test\runner.vcxproj] libpqxx-7.9.a(wait.obj) : error LNK2019: unresolved external symbol __imp_WSAPoll referenced in function "void __cdecl pqxx::internal::wait_fd(int,bool,bool,unsigned int,unsigned int)" (?wait_fd@internal@pqxx@@YAXH_N0II@Z) [C:\Libraries\l ibpqxx\build_arm\test\runner.vcxproj] C:\Libraries\libpqxx\build_arm\test\Release\runner. : fatal error LNK1120: 3 unresolved externals [C:\Libraries\libpqxx \build_arm\test\runner.vcxproj]

i tried to change the CMakeLists and everything, but nothing happened.

i tried the instructions so i can include the library into my plc next project, because i cant build my project without the proper library.

**Comment by tt4g** [comment]
#865

**Comment by jtv** [comment]
@fuadsmed was this ever resolved?

**Reply by ghost** [reply]
No, unfortunately I can't find a solution for this :(

**Reply by jtv** [reply]
Have you looked at #381?  It sounds like a missing library.

**Reply by jtv** [reply]
And #438 looks like it might be related.

**Reply by jtv** [reply]
And I don't suppose it's [this problem](https://stackoverflow.com/questions/37714393/visual-studio-2015-community-zeromq-static-library-linking-error) but can't hurt to check...
